adobe/react-spectrum
# Adding an option to control if onSelectionChange fires when the selected key doesn't change

Closes #2536

## ✅ Pull Request Checklist:


## 📝 Test Instructions:

1. Go to ListView/TableView highlight selection + onAction stories.
2. Verify that clicking/pressing Enter/Space on a selected row doesn't fire additional onSelectionChange events. Enter should fire onAction
3. Go to Tabs and click on the same tab multiple times. Verify that onSelectionChange fires multiple times with the same key
4. Go to Picker/ComboBox. Select an option, then reopen the menu and reselect the same option. onSelectionChange should fire w/ the same key and the menu should close
5. Try other components that support selection and verify that they work as expected (Menu, ListBox, etc)

https://reactspectrum.blob.core.windows.net/reactspectrum/992beb53ac52646a1d88e3b3ae32f113769fe329/storybook/index.html?path=/story/tableview--selectionstyle-highlight-onaction is an older storybook to compare behavior with. Everything should behave the same except for ListView/TableView highlight selection and ActionGroup disallowEmptySelection story should only fire onSelectionChange when the selected button changes

## 🧢 Your Project:

RSP
